src/react/hooks/__tests__/useSubscription.test.tsx

```diff
@@ -1122,6 +1122,276 @@ followed by new in-flight setup", async () => {
   });
 });
 
+describe("ignoreResults", () => {
+  const subscription = gql`
+    subscription {
+      car {
+        make
+      }
+    }
+  `;
+
+  const results = ["Audi", "BMW"].map((make) => ({
+    result: { data: { car: { make } } },
+  }));
+
+  it("should not rerender when ignoreResults is true, but will call `onData` and `onComplete`", async () => {
+    const link = new MockSubscriptionLink();
+    const client = new ApolloClient({
+      link,
+      cache: new Cache({ addTypename: false }),
+    });
+
+    const onData = jest.fn((() => {}) as SubscriptionHookOptions["onData"]);
+    const onError = jest.fn((() => {}) as SubscriptionHookOptions["onError"]);
+    const onComplete = jest.fn(
+      (() => {}) as SubscriptionHookOptions["onComplete"]
+    );
+    const ProfiledHook = profileHook(() =>
+      useSubscription(subscription, {
+        ignoreResults: true,
+        onData,
+        onError,
+        onComplete,
+      })
+    );
+    render(<ProfiledHook />, {
+      wrapper: ({ children }) => (
+        <ApolloProvider client={client}>{children}</ApolloProvider>
+      ),
+    });
+
+    const snapshot = await ProfiledHook.takeSnapshot();
+    expect(snapshot).toStrictEqual({
+      loading: false,
+      error: undefined,
+      data: undefined,
+      variables: undefined,
+    });
+    link.simulateResult(results[0]);
+
+    await waitFor(() => {
+      expect(onData).toHaveBeenCalledTimes(1);
+      expect(onData.mock.lastCall?.[0].data).toStrictEqual({
+        data: results[0].result.data,
+        error: undefined,
+        loading: false,
+        variables: undefined,
+      });
```

Comment:
```suggestion
      expect(onData).toHaveBeenLastCalledWith(
        expect.objectContaining({
          data: {
            data: results[0].result.data,
            error: undefined,
            loading: false,
            variables: undefined,
          },
        })
      );
```

[nit] As a readability thing, it would be great if we could structure this using the expect helpers (`toHaveBeenLastCalledWith`) rather than trying to directly read from `mock` directly. I think you were looking to avoid checking against every property passed to that first argument, so perhaps using `expect.objectContaining` can also get around that. I'm able to get the test to pass the same structuring it this way. This also makes it clear to the reader that there are more properties passed to this argument that we aren't checking as a part of this test 🙂 

(this suggestion applies to any location with `.mocks.lastCall`, so I'll avoid adding this to each place)

Reply:
Fair enough - I think I had tried that before, but probably missed something :)

Reply:
[2a7f228](https://github.com/apollographql/apollo-client/pull/11921/commits/2a7f228da3fc5c4c8f0f80f75e91dae021c57647)